pta-intelligent-search-api/src/main/java/fi/maanmittauslaitos/pta/search/api/region/RegionNameSearchResult.java

```diff
@@ -0,0 +1,41 @@
+package fi.maanmittauslaitos.pta.search.api.region;
+
+import fi.maanmittauslaitos.pta.search.api.Language;
+
+import java.util.Map;
+
+public class RegionNameSearchResult {
+    private boolean hasRegionName;
+    private String regionalQueryPart;
+    private String parsedRegion;
+
+    public static RegionNameSearchResult executeSearch(String queryTerm, String stemmedQueryTerm, RegionNameContainer regionNameContainer, Language lang) {
+        Map<String, String> stemmedRegionNames = regionNameContainer.getStemmedRegionNames().get(lang);
+        if (stemmedRegionNames.containsKey(stemmedQueryTerm)) {
+            return new RegionNameSearchResult(true, queryTerm, stemmedRegionNames.get(stemmedQueryTerm));
+        }
+        return noRegionFound();
+    }
+
+    public static RegionNameSearchResult noRegionFound() {
+        return new RegionNameSearchResult(false, "", "");
```

Comment:
As the objects are immutable, the "no region found" object could be a final static variable